backend/cfg/cfg_to_linear.ml

```diff
@@ -308,11 +308,21 @@ let adjust_stack_offset body (block : Cfg.basic_block)
     let delta_bytes = block_stack_offset - prev_stack_offset in
     to_linear_instr (Ladjust_stack_offset { delta_bytes }) ~next:body
 
+let array_of_layout : Cfg_with_layout.layout -> Label.t array =
```

Comment:
We don't need to create an array here. It is only used to make it easier to traverse the list backwards and have access to both layout.(i) and layout.(i-1) in the same iteration.  It should be easy to implement with DoublyLinkedList using `prev` links. I think it is worth fixing, as it will make the code a little more efficien, with a relatively small extension of the interface of DoublyLinkedList.

Reply:
@xclerc   what do you think about this version: https://github.com/ocaml-flambda/flambda-backend/pull/1124/commits/38623bb00f60a1da37e51ac7468d1eb7be61783b